azure-configurationservice/azure/configuration/aio/azure_configuration_client_async.py

```diff
@@ -0,0 +1,262 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+import os
+import sys
+import platform
+from datetime import datetime
+from msrest.paging import Paged
+from azure.core.pipeline import AsyncPipeline
+from azure.core.pipeline.transport import AsyncioRequestsTransport
+from azure.core.exceptions import ResourceModifiedError, ResourceNotFoundError
+from ..utils import get_endpoint_from_connection_string
+from .._generated.aio import AzureConfigurationClientImp
+from .._generated.aio import AzureConfigurationClientImpConfiguration
+from ..azure_configuration_requests import AzConfigRequestsCredentialsPolicy
+from ..azure_configuration_client_abstract import AzureConfigurationClientAbstract
+from .._generated.models import ConfigurationSetting
+
+
+class AzureConfigurationClient(AzureConfigurationClientAbstract):
+    """
+    This is the async version of :class:`azure.configuration.ConfigurationClient`
+
+    Example
+
+    .. code-block:: python
+
+        from azure.configuration.aio import AzureConfigurationClient
+        connection_str = "<my connection string>"
+        async_client = AzureConfigurationClient(connection_str)
+    """
+
+    def __init__(self, connection_string):
```

Comment:
The constructor should take an options/configuration object that allows for control of the pipeline creation.

Reply:
Will update this

Reply:
backlog created. Resove this one for now